cookbook/bundles/tests.rst

```diff
@@ -0,0 +1,311 @@
+
+        # Tests/app/config.yml
+        framework:
+            secret: Th1s1sS3cr3t!
+
+Symfony applications generate cache and logs directories, which can be ignored by
+adding those lines to ``.gitignore``:
+
+.. code-block::
+
+    # .gitignore
+    /Tests/app/cache
+    /Tests/app/logs
+    /vendor
+    /composer.lock
+
+Browsing Pages
+--------------
+
+If the third party bundle provides a controller, it can be helpful to create a
+front controller similar to the one that can be found in Symfony's Standard Edition::
+
+    // Tests/app/web.php
+    use Symfony\Component\HttpFoundation\Request;
+
+    require_once __DIR__.'/autoload.php';
+
+    $kernel = new AppKernel('prod', false);
+    $request = Request::createFromGlobals();
+    $response = $kernel->handle($request);
+    $response->send();
+
+With this it becomes possible to browse the page. A way to do it without having to configure
+a web server is to run the following command:
+
+.. code-block:: bash
+
+    $ php Tests/app/console.php server:run -d Tests/app
+
+.. note::
+
+    If the third party bundle uses the Twig templating engine to render HTML pages
+    or if it uses the Form Component or anything else, then more dependencies and
+    configuration parameters should be added.
+
+Running Commands
+----------------
+
+If the third party bundle provides a command, it can be helpful to create a ``console``
+similar to the one that can be found in Symfony's Standard Edition::
+
+    // Tests/app/console.php
+    set_time_limit(0);
+
+    require_once __DIR__.'/autoload.php';
+
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+
+    $kernel = new AppKernel('dev', true);
+    $application = new Application($kernel);
+    $application->run();
+
+With this it becomes possible to run manually the console:
+
+.. code-block:: bash
+
+    $ php Tests/app/console.php
+
+Automated Tests
+---------------
+
+Manual tests are great to get a quick idea of what the bundle does. But writing
+automated tests is even better!
+
+The first step is to install a test framework like PHPUnit:
+
+.. code-block:: bash
+
+    $ composer require --dev phpunit/phpunit
+
+.. note::
+
+    The steps should be similar with other tests frameworks.
+
+Then the second one is to configure it to use Composer's autoloading:
+
+.. configuration-block::
+
+    .. code-block:: xml
+
+        <?xml version="1.0" encoding="UTF-8"?>
+        <!-- phpunit.xml.dist -->
+
+        <!-- http://phpunit.de/manual/4.3/en/appendixes.configuration.html -->
+        <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.3/phpunit.xsd"
+            backupGlobals="false"
+            colors="true"
+            bootstrap="./vendor/autoload.php"
+        >
+            <testsuites>
+                <testsuite name="Test Suite">
+                    <directory>./Tests/</directory>
+                </testsuite>
+            </testsuites>
+
```

Comment:
It's only autoloaded if you use `KernelTestCase` with PHPUnit.
I personally prefer to explicitly load `AppKernel` as it will work with any test framework.